### PR TITLE
implement max payments

### PR DESCRIPTION
**Pull request by abhay**

### CODE REVIEW

**Comment by abhay** [comment]
Something like this should work for implementing max payments. Need someone to pick up with WIP.

File: src/bn_wallets.erl [hunk 1]
```diff
@@ -327,11 +328,16 @@ jsonrpc_nonce_param(Param, Address, NonceType, Chain) ->
 ) ->
     {ok, blockchain_txn:txn()} | {error, term()}.
 mk_payment_txn_v2(Payer, PaymentList, Nonce, Chain) ->
-    Payments = [blockchain_payment_v2:new(Payee, Bones) || {Payee, Bones} <- PaymentList],
+    Payments = [mk_payment(Payee, Bones, Max) || {Payee, Bones, Max} <- PaymentList],
     Txn = blockchain_txn_payment_v2:new(Payer, Payments, Nonce),
     TxnFee = blockchain_txn_payment_v2:calculate_fee(Txn, Chain),
     {ok, blockchain_txn_payment_v2:fee(Txn, TxnFee)}.
 
+mk_payment(Payee, _Bones, true) ->
```
Comment (abhay) [inline src/bn_wallets.erl]:
```suggestion
mk_payment(Payee, undefined, true) ->
```

File: src/bn_wallets.erl [hunk 2]
```diff
@@ -327,11 +328,16 @@ jsonrpc_nonce_param(Param, Address, NonceType, Chain) ->
...
-    Payments = [blockchain_payment_v2:new(Payee, Bones) || {Payee, Bones} <- PaymentList],
+    Payments = [mk_payment(Payee, Bones, Max) || {Payee, Bones, Max} <- PaymentList],
     Txn = blockchain_txn_payment_v2:new(Payer, Payments, Nonce),
     TxnFee = blockchain_txn_payment_v2:calculate_fee(Txn, Chain),
     {ok, blockchain_txn_payment_v2:fee(Txn, TxnFee)}.
 
+mk_payment(Payee, undefined, true) ->
+    blockchain_payment_v2:new(Payee, max);
+mk_payment(Payee, Bones, false) ->
+    blockchain_payment_v2:new(Payee, Bones).
```
Comment (abhay) [inline src/bn_wallets.erl:353]: nb, this probably needs some validation work or a better error message but this will probably do the job immediately.

**Comment by jeffgrunewald** [comment]
@abhay @joecaswell and i should have this ready to go; he spotted and added the places that would need to accept the new payment parameter and i added some error handling to return a jsonrpc error to the caller if the combination of payment arguments submitted is invalid (combination of a max payment and a concrete value in bones)

**Review by madninja — APPROVED** [review]

**Pull request merged**